site/gatsby-node.js

```diff
@@ -22,22 +39,134 @@ module.exports.onCreateNode = ({ node, actions, getNode }) => {
         node,
         value: sourceInstanceName,
       })
+
+      actions.createNodeField({
+        name: 'version',
+        node,
+        value: protocolVersion,
+      })
+
+      actions.createNodeField({
+        name: 'avatar',
+        node,
+        value: avatar,
+      })
+
+      actions.createNodeField({
+        name: 'modifiedTime',
+        node,
+        value: new Date(fileNode.modifiedTime),
+      })
     }
   }
 }
 
 module.exports.createSchemaCustomization = ({ actions }) => {
   const { createTypes } = actions
-  // TODO add Frontmatter types (title, tags etc)
+
   const typeDefs = `
-    type MarkdownRemark implements Node @dontInfer {
-        fields: MarkdownRemarkFields
+  type MarkdownRemark implements Node {
+      frontmatter: Frontmatter
+      fields: Fields
     }
- 
-    type MarkdownRemarkFields {
+    type Frontmatter {
+      title: String
+      tags: [String]
+      licence: String
+      username: String
+      status: String
+    }
+    type Fields {
       collection: String
       slug: String
+      version: String
+      avatar: String
+      modifiedTime: Date
     }
   `
   createTypes(typeDefs)
 }
+
+module.exports.createPages = async ({ actions, graphql }) => {
+  const allMarkdown = await graphql(`
+    {
+      protocols: allMarkdownRemark(filter: { fields: { collection: { eq: "protocols" } } }) {
+        nodes {
+          id
+          fields {
+            slug
+            version
+            avatar
+            modifiedTime
+          }
+          frontmatter {
+            title
+            tags
+            licence
+            username
+            status
+            summary
+          }
+        }
+      }
+    }
+  `)
+  if (allMarkdown.errors) {
+    allMarkdown.errors.forEach((e) => console.error(e.toString()))
+    throw Error(allMarkdown.errors)
+  }
+
+  const protocolTemplate = path.resolve(`src/templates/Protocol/Protocol.tsx`)
+  const pages = allMarkdown.data.protocols.nodes
+  pages.forEach((page) => {
+    const { id, fields } = page
+    console.log('created ' + fields.slug)
+    actions.createPage({
+      path: fields.slug,
+      component: protocolTemplate,
+      context: { id },
+    })
+  })
+
+  createSearchPage(actions.createPage, allMarkdown.data.protocols.nodes)
+}
+
+const normalizeProtocol = (node) => ({
```

Comment:
Can you don't define a lambda function here? It uses only in one place

Reply:
fixed